pkg/git/error_utils.go

```diff
@@ -0,0 +1,44 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type gitError struct {
+	msg string
+}
+
+// GitError returns an error whereby the user part of the url is removed (e.g. an access token)
+// Accepts the message to use and the url to remove the user part from
+func GitError(msg, url string) error {
+	cleanedURL, errorFromClean := cleanURL(url)
+	if errorFromClean != nil {
+		return errorFromClean
+	}
+	return gitError{msg: fmt.Sprintf("%s. repository URL = %s", msg, cleanedURL)}
+}
+
+// IsGitError checks if the error is a Git one (typically we see status code 128 and git in the string)
+func IsGitError(e error) bool {
+	// This is the most common we've seen so far, todo augment this as we noticed more to handle
+	if strings.Contains(e.Error(), "exit status 128") && strings.Contains(e.Error(), "git") {
+		return true
+	}
+	return false
+}
+
+func cleanURL(inputURL string) (string, error) {
+	parsedURL, parseErr := url.Parse(inputURL)
+	if parseErr != nil {
+		return "", errors.New("failed to parse repository URL")
+	}
+	parsedURL.User = nil
+	return fmt.Sprintf("%s", parsedURL), nil
```

Comment:
you can simplify this a bit:

```suggestion
	return parsedURL.String(), nil
```